src/lava/magma/runtime/runtime_service.py

```diff
@@ -116,7 +150,26 @@ def _get_pm_resp(self) -> ty.Iterable[MGMT_RESPONSE]:
             ptos_recv_port = self.process_to_service[counter]
             rcv_msgs.append(ptos_recv_port.recv())
             counter += 1
-        return rcv_msgs
+        for recv_msg in rcv_msgs:
```

Comment:
Might not be the most performant thing to do. If in one time step some processes request LRN and others request PRE_MGMT then both request flags will be set. _next_phase() will make sure that we execute PRE_MGMT next. 
in the next phase we don't even have to check any of the rcv_msgs again if req_lrn is already set.
Same holds for other phases...